pono.cpp

```diff
@@ -102,6 +107,29 @@ ProverResult check_prop(PonoOptions pono_options,
   return r;
 }
 
+// Note: signal handlers are registered only when profiling is enabled.
```

Comment:
I don't know as much about signal handlers as I should. Is there a big performance overhead? If not, maybe we should just register them always? Maybe it would give a more graceful exit. Although I guess I'm not sure exactly what we would do in the handler when we're not profiling.